roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/event/CarCouldLeaveSectionEvent.java

```diff
@@ -78,7 +77,7 @@ public void eventRoutine(Street donorSection) throws SuspendExecution {
                     new TimeSpan((donorSection.getFirstCar().getTimeToTraverseSection(nextSection)), TimeUnit.SECONDS)
                 );
                 donorSection.moveFirstCarToNextSection();
-            } else if (nextSection != null && (nextSection instanceof RoundaboutSink || nextSection instanceof OneWayStreetAdapter)) {
+            } else if (nextSection != null && (nextSection instanceof RoundaboutSink || nextSection instanceof Intersection)) {
```

Comment:
not sure if this is done somewhere else or if i missunderstand something:
what if the nextsection is a street(section)?

Reply:
this is from @sro1993 branch